src/app/api/routes/devices.py

```diff
@@ -13,29 +13,32 @@
     DeviceOut,
     DeviceAuth,
     MyDeviceAuth,
+    AdminDeviceAuth,
     DeviceCreation,
     DeviceIn,
     UserRead,
     DefaultPosition,
     Cred,
-    AccessType
+    AccessType,
 )
 from app.api.deps import get_current_device, get_current_user
+from app.api.crud.groups import get_entity_group_id
 
 
 router = APIRouter()
 
 
 @router.post("/", response_model=DeviceOut, status_code=201, summary="Create a new device")
-async def register_device(payload: DeviceAuth, _=Security(get_current_user, scopes=[AccessType.admin])):
+async def register_device(payload: AdminDeviceAuth, _=Security(get_current_user, scopes=[AccessType.admin])):
     """Creates a new device based on the given information
 
     Below, click on "Schema" for more detailed information about arguments
     or "Example Value" to get a concrete idea of arguments
     """
     if await crud.get(payload.owner_id, users) is None:
         raise HTTPException(status_code=404, detail=f"Unknown user for owner_id={payload.owner_id}")
-    return await crud.accesses.create_accessed_entry(devices, accesses, payload, DeviceCreation)
+    device_payload = DeviceAuth(**payload.dict(), group_id=await get_entity_group_id(users, payload.owner_id))
```

Comment:
I suggest setting the owner's group id only if the admin didn't specify it in the payload

Reply:
Done but does it make sens to have a group_id different from the group of the owner?

Reply:
For the admin, I think it does: if a user isn't able to register her/his device, the admin will be able to do so while registering the correct group